src/update-process-env.js

```diff
@@ -0,0 +1,49 @@
+/** @babel */
+
+import {spawnSync} from 'child_process'
+
+const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set(['NODE_ENV', 'NODE_PATH'])
+
+export default function updateProcessEnv (launchEnv) {
+  let envToAssign
+  if (launchEnv.PWD) {
+    envToAssign = launchEnv
+  } else {
+    if (process.platform === 'darwin') {
+      envToAssign = getEnvFromShell()
+    }
+  }
+
+  if (envToAssign) {
+    for (let key in process.env) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        delete process.env[key]
+      }
+    }
+
+    for (let key in envToAssign) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+        process.env[key] = envToAssign[key]
+      }
+    }
+  }
+}
+
+function getEnvFromShell () {
```

Comment:
@steelbrain As someone who is ultimately responsible for maintaining this behavior core, I find it easier to be able to browse to a specific chunk of code that performs this logic and immediately see what's going on rather than needing to dig it up in a dependency. It just doesn't seem complex enough to warrant it. That isn't always the case, but for this concern it is. It's a subjective call but that's my preference.

I would be very interested in accepting a specific PR addressing specific issues in the current code, however.